src/tools/rbd_mirror/group_replayer/Replayer.cc

```diff
@@ -467,6 +467,29 @@ void Replayer<I>::handle_load_remote_group_snapshots(int r) {
   if (is_replay_interrupted()) {
     return;
   }
+  if (!r) {
+    for (auto snap = m_remote_group_snaps.begin(); snap!= m_remote_group_snaps.end();) {
+      auto ns = std::get_if<cls::rbd::GroupSnapshotNamespaceMirror>(&snap->snapshot_namespace);
+      if (ns == nullptr) {
+        ++snap;
+        continue;
+      }
+      if (ns->mirror_peer_uuids.count(m_local_mirror_uuid) == 0) {
+        dout(10) << "Skipping snapshot " << snap->id
+                << " with no matching mirror_peer_uuid in: "
+                << ns->mirror_peer_uuids << " (expected: " << m_local_mirror_uuid << ")"
+                << dendl;
+        snap = m_remote_group_snaps.erase(snap);
```

Comment:
This check should probably be done in scan_for_unsynced_group_snapshots(). Also, do not erase any entries from the snap lists.